app/src/main/java/org/schabi/newpipe/database/history/dao/SearchHistoryDAO.java

```diff
@@ -36,16 +36,16 @@
     @Override
     Flowable<List<SearchHistoryEntry>> getAll();
 
-    @Query("SELECT * FROM " + TABLE_NAME + " GROUP BY " + SEARCH + ORDER_BY_CREATION_DATE
+    @Query("SELECT DISTINCT " + SEARCH + " FROM " + TABLE_NAME + ORDER_BY_CREATION_DATE
             + " LIMIT :limit")
-    Flowable<List<SearchHistoryEntry>> getUniqueEntries(int limit);
+    Flowable<List<String>> getUniqueEntries(int limit);
 
     @Query("SELECT * FROM " + TABLE_NAME
             + " WHERE " + SERVICE_ID + " = :serviceId" + ORDER_BY_CREATION_DATE)
     @Override
     Flowable<List<SearchHistoryEntry>> listByService(int serviceId);
 
-    @Query("SELECT * FROM " + TABLE_NAME + " WHERE " + SEARCH + " LIKE :query || '%'"
-            + " GROUP BY " + SEARCH + " LIMIT :limit")
-    Flowable<List<SearchHistoryEntry>> getSimilarEntries(String query, int limit);
+    @Query("SELECT DISTINCT " + SEARCH + " FROM " + TABLE_NAME
+            + " WHERE " + SEARCH + " LIKE :query || '%'" + ORDER_BY_CREATION_DATE + " LIMIT :limit")
```

Comment:
`` || ``

Shouldn't that be an "OR" - we are in SQL not Java :D

Reply:
That's string concatenation

Reply:
Oh wow I didn't know that.
Normal SQL-languages use "CONCAT" or "+" for that but sqlite seems to be special in that case...